### DISCUSSION TITLE
Map content discrepancy between dashboard map and logbook map

### DISCUSSION THREAD

**Discussion by filipmelik** [wavelog/wavelog#1410] [Ideas]
Hello,

1] since first weeks of usage of the wavelog, there is this thing that is bugging me: Is there a reason why the map on the dashboard and map on the logbook page show different data? I know this is documented: `The map only shows contacts made in the last 7 days, on the logbook page it shows the contacts listed on that page.`, so it looks like a on-purpose feature, but to be honest, I find this very unintuitive. This gets especially confusing in case you have no QSO for a while, then the map is empty, but the list of QSOs below is not.

What I am expecting as an user, is that the map shows the QSOs listed in the table below the map, in same fashion, as on the Logbook page.

2] Second thing I would like to discuss, in conjunction with the above, is the number of QSOs shown on the dashboard. Apparently, the number is hardcoded to 18, [as seen here](https://github.com/wavelog/wavelog/blob/c0fd01e0010fd63fde9efd4efd341df34e11fba1/application/controllers/Dashboard.php#L118C10-L118C24). 
The immediate thing that came to my mind is "why 18? why not 5, 15, x?" 😄 On this one, I have two ideas that we could employ:

- either make this number configurable (that involves new user setting, migration etc. - more work to do)
- or we could add this cool note, that is visible on QSO page, as seen on this screenshot (easier to do. we could even do this change in combination with above):

<img width="805" alt="Screenshot 2024-12-30 at 11 34 58" src="https://github.com/user-attachments/assets/39a6aceb-3cc3-46c8-8679-8753a7a445e2" />

But this "number of QSOs" is more like a bonus, as I am especially concerned with the dashboard's map behavior. WDYT? 😊

**Comment by int2001** [comment]
just to be clear in case of misunderstandings:

if one implements a configurable attribute (like this one), it is necessary to implement a user or admin-configurable item as well.
Some const which has to be changed at some php-script isn't acceptable. in some rare-cases we put __global__ config-items into config.php. but amount of shown QSOs shouldn't be a global config.

For this one we'd suggest a user-option at the UI (See User-Controller and Views) which can be easily stored and retrieved into/from user_options.

**Reply by filipmelik** [reply]
So the final conclusion about this matter is that the number of displayed QSOs in **Dashboard** should be configurable using user-settings, do I understand it correctly? Just double checking if we are all on the same boat here, so I can continue with implementation 😊

**Reply by int2001** [reply]
yep.
